src/runtime/merge.rs

```diff
@@ -6,96 +6,77 @@ use std::{
     task::{Context, Poll},
 };
 
-use futures::{stream::Fuse, Stream, StreamExt};
+use futures::{Stream, StreamExt};
 
 /// Provides the ability to merge two streams, exiting when either stream is exhausted.
 pub(crate) trait MergeExt: Stream {
-    fn merge<S>(self, other: S) -> Merge<Self, S>
+    fn merge<S>(self, other: S) -> Merge<Self::Item>
     where
-        Self: Sized,
-        S: Stream<Item = Self::Item>,
+        Self: 'static + Sized + Send,
+        S: 'static + Stream<Item = Self::Item> + Send,
     {
-        Merge::new(self, other)
+        Merge::new(self).merge(other)
     }
 }
 impl<S: ?Sized> MergeExt for S where S: Stream {}
 
 /// Stream returned by the [`merge`](MergeExt::merge) method.
 #[pin_project::pin_project]
-pub(crate) struct Merge<T, U> {
+pub(crate) struct Merge<I> {
```

Comment:
You are generalizing this whole Stream `Merge` logic to merge multiple streams, right? This feels a bit like a separate feature PR.